ui/src/DataService.js

```diff
@@ -10,28 +18,59 @@ const requestConfig = (method, body) => ({
   body: body && JSON.stringify(body)
 });
 
+/**
+ * Sends a HTTP request
+ *
+ * @param {String} path - The URI to send the request to
+ * @param {*} method - HTTP Method (PUT, GET, PATCH, DELETE, POST)
+ * @param {*} body - The request payload
+ * @returns {Boolean|JSON} - Returns the response from the HTTP request
+ */
 const request = async (path, method, body) => {
   const response = await fetch(`${config.api.url}/${path}`, requestConfig(method, body));
   if (!response.ok) {
     const msg = await response.text();
-    throw Error(msg);
+    throw Error(`${response.statusText}: ${msg}`);
   }
-  return method === 'DELETE' || response.json();
+
+  return (method === 'DELETE' || response.status === 204) || response.json();
 };
 
-const fetchItems = async path => {
-  const result = await request(path, 'GET');
+/**
+ * Makes a HTTP GET request to a provided URI and returns an array of items
+ *
+ * @param {String} url - The full URL to make the GET request to
+ */
+const fetchItems = async url => {
+  const result = await request(url, 'GET');
   return result || [];
 };
 
+/**
+ * Makes a HTTP GET request to a provided URI and returns a single item
+ *
+ * @param {String} url - The full URL to make the GET request to
+ */
 const fetchItem = async url => {
   const result = await request(url, 'GET');
   return result || {};
 };
 
+/**
+ * Update a number of app versions by their app ID
+ *
+ * @param {String} id - The App ID related of the versions
+ * @param {Array} versions - The list of versions to update
+ */
+const updateAppVersions = async (id, versions) => {
+  const result = request(`apps/${id}/versions`, 'PUT', versions);
+  return result || [];
+};
+
 const dataService = {
   fetchApps: () => fetchItems('apps'),
-  getAppById: (id) => fetchItem(`apps/${id}`)
+  getAppById: (id) => fetchItem(`apps/${id}`),
+  updateAppVersions: (id, versions) => updateAppVersions(id, versions)
```

Comment:
this could be shortened to 
```suggestion
  updateAppVersions
```